src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -202,11 +211,29 @@ public DiagnosticResult EmitAssembly(string outputPath)
 
                 Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
 
+                var diagnostics = new List<Diagnostic>(emitResult.Diagnostics);
+
+                if (CompilationContext.DiagnosticAnalyzers != null && CompilationContext.DiagnosticAnalyzers.Length > 0)
+                {
+                    sw.Start();
+
+                    var analyzerDiagnostics = CompilationContext.Compilation
+                        .WithAnalyzers(CompilationContext.DiagnosticAnalyzers)
+                        .GetAllDiagnosticsAsync()
+                        .Result     // TODO: make the caller an async method?
```

Comment:
Can you either a) try making it async or b) file a bug to investigate this further?